Std/Data/Bool.lean

```diff
@@ -38,6 +38,50 @@ instance (x y : Bool) : Decidable (x < y) := inferInstanceAs (Decidable (!x && y
 instance : Max Bool := ⟨or⟩
 instance : Min Bool := ⟨and⟩
 
+theorem false_ne_true : false ≠ true := Bool.noConfusion
+
+theorem eq_false_or_eq_true : (b : Bool) → b = true ∨ b = false := by decide
+
+theorem eq_false_iff : {b : Bool} → b = false ↔ b ≠ true := by decide
+
+theorem eq_true_iff : {b : Bool} → b = true ↔ b ≠ false := by decide
+
+/-! ### beq -/
+
+@[simp] theorem beq_false_left : ∀ (x : Bool), (false == x) = !x := by decide
+
+@[simp] theorem beq_false_right : ∀ (x : Bool), (x == false) = !x := by decide
+
+@[simp] theorem beq_true_left : ∀ (x : Bool), (true == x) = x := by decide
+
+@[simp] theorem beq_true_right : ∀ (x : Bool), (x == true) = x := by decide
+
+theorem beq_not_left : ∀ (x y : Bool), ((!x) == y) = !(x == y) := by decide
+
+theorem beq_not_right : ∀ (x y : Bool), (x == (!y)) = !(x == y) := by decide
+
+@[simp] theorem beq_not_not : ∀ (x y : Bool), ((!x) == (!y)) = (x == y) := by decide
+
+theorem beq_eq_not_xor : ∀ (x y : Bool), (x == y) = !xor x y := by decide
+
+/-! ### bne -/
+
+theorem bne_false_left : ∀ (x : Bool), (false != x) = x := by decide
+
+theorem bne_false_right : ∀ (x : Bool), (x != false) = x := by decide
+
+theorem bne_true_left : ∀ (x : Bool), (true != x) = !x := by decide
+
+theorem bne_true_right : ∀ (x : Bool), (x != true) = !x := by decide
+
+theorem bne_not_left : ∀ (x y : Bool), ((!x) != y) = (x == y) := by decide
+
+theorem bne_not_right : ∀ (x y : Bool), (x != (!y)) = (x == y) := by decide
+
+theorem bne_not_not : ∀ (x y : Bool), ((!x) != (!y)) = (x != y) := by decide
+
+theorem bne_eq_xor : ∀ (x y : Bool), (x != y) = xor x y := by decide
```

Comment:
There are so many potential rules here and normal forms may be computationally intractable.

Is there a criteria used for these rules in particular?

Reply:
Currently `xor` is an alias of `bne`, so its normal form is the same as `xor`.

Reply:
Since `xor` is an abbreviation for `bne`, simp sees through the abbreviation and all simp lemmas for `xor` work for `bne`. However, the seemingly duplicate lemmas are useful for `rw` which doesn't see through abbreviations as well. Because none of the proposed `bne` lemmas are marked simp there is no problem with adding these.

There is a bigger issue though about the simp normal form for `Bool`. Since `x != y` is `!( x == y)`, there is a chance that simp might run into issues where `==` leads one way but `xor` leads another way. I think that figuring out what the right simp normal form for `Bool` can wait a bit though and shouldn't bog down this PR.

Reply:
Given all the `xor` theorems, the `bne` theorems feel unnecessary.  I'd suggest removing the bne theorems or if there are ones not found in the xor theorems replacing bne with xor.